Mathlib/Algebra/Category/Ring/Basic.lean

```diff
@@ -136,6 +136,8 @@ def _root_.RingEquiv.toSemiRingCatIso [Semiring X] [Semiring Y] (e : X ≃+* Y)
     SemiRingCat.of X ≅ SemiRingCat.of Y where
   hom := e.toRingHom
   inv := e.symm.toRingHom
+  hom_inv_id := by ext; erw [comp_apply, id_apply, e.symm_apply_apply]
+  inv_hom_id := by ext; erw [comp_apply, id_apply, e.apply_symm_apply]
```

Comment:
:-(

Reply:
It's because `symm_apply_apply`

Reply:
How would you feel about:
```
@[simp]
lemma foo [Semiring X] [Semiring Y] (e : X ≃+* Y) :
  ↑(RingEquiv.toRingHom e) = ↑e := rfl

@[simp]
lemma bar [Semiring X] [Semiring Y] (e : X ≃+* Y) :
    (@FunLike.coe (of X ⟶ of Y) _ (fun _ => (forget SemiRingCat).obj _)
      ConcreteCategory.funLike (e : X →+* Y) : X → Y) = ↑e :=
  rfl

@[simps]
def _root_.RingEquiv.toSemiRingCatIso [Semiring X] [Semiring Y] (e : X ≃+* Y) :
    SemiRingCat.of X ≅ SemiRingCat.of Y where
  hom := (e : X →+* Y)
  inv := (e.symm : Y →+* X)
```

Reply:
Unfortunately `bar` would need to come in separate versions also for `Ring` and `CommRing`.

Reply:
`foo` is already in mathlib4 under
```lean
@[simp, norm_cast]
theorem coe_toRingHom (f : R ≃+* S) : ⇑(f : R →+* S) = f :=
  rfl
#align ring_equiv.coe_to_ring_hom RingEquiv.coe_toRingHom
```

Let me make `bars`

Reply:
While making `bars`, I find that `CommSemiRing` is not named `CommSemiRingCat` which I think is an unintentional mistake, I am going to make another PR to fix this